torch/_dynamo/variables/lists.py

```diff
@@ -25,27 +25,47 @@
 from .functions import UserFunctionVariable, UserMethodVariable
 
 
-def _listlike_contains_helper(items, search, tx, options):
+def _listlike_contains_helper(items, search, tx, options, check_hash_match=False):
```

Comment:
Why adding the `check_hash` as input, rather than always checking?

Reply:
Because the behaviour for list is not to check object identity. 

Furthermore, we only `check_hash_match` for `Tensor` since we only know for a fact that tensor's hash is id and hence equivalent to its object identity.

We let the rest fail.

Reply:
fair enough. All this will be properly fixed in https://github.com/pytorch/pytorch/pull/110524, but this LGTM as a workaround

Reply:
We could extend this to `numpy.NdArray` if you would like @lezcano . Perhaps in follow-up PR.